.github/actions/build-src/action.yaml

```diff
@@ -0,0 +1,69 @@
+name: 'build-src'
+description: 'make source builds for CI'
+inputs:
+  build-hole:
+    description: 'build HOLE2'
+    required: true
+    default: true
+  build-tests:
+    description: 'build MDA tests'
+    required: true
+    default: true
+  build-docs:
+    description: 'build MDA docs'
+    required: true
+    default: false
+
+
+runs:
+  using: "composite"
+  steps:
+    - name: echo_inputs
+      shell: bash -l {0}
+      run: |
+        echo ${{ inputs.build-hole }}
+        echo ${{ inputs.build-tests }}
+        echo ${{ inputs.build-docs }}
+
+    - name: check_setup
+      shell: bash -l {0}
+      run: |
+        # Check OS and python setup prior to builds
+        echo "OS: ${OS_NAME}"
+        which python
+        which pip
+        pip list
+        conda info
+        conda list
+
+    - name: build_hole
+      if: ${{ inputs.build-hole == 'true' }}
+      shell: bash -l {0}
+      run: |
+        # We manually build hole2 to avoid OS incompatibilities
+        git clone https://github.com/MDAnalysis/hole2.git
+        cd hole2/src
+        source ../source.apache
+        (make FC=${FC}) && (make PREFIX=${HOME}/hole2 FC=${FC} install)
+        source ../source.unset
+        echo "HOLE_BINDIR=${HOME}/hole2/bin" >> $GITHUB_ENV
+        echo "${HOME}/hole2/bin" >> $GITHUB_PATH
+
+    - name: build_mda_main
+      shell: bash -l {0}
+      run: |
+        # TODO: using install instead of develop here causes coverage to drop
+        # for .pyx file. If possible a solution for this should be found.
+        cd package/ && python setup.py develop
```

Comment:
I think this is because with pyx files the pyx source files don't get put into installation (as they become .so and unrecognisable?), whereas a develop install leaves everything "in tree" and more visible to coverage.  Is there a specific problem with a develop install?

Reply:
Not that I'm aware of, this was a TODO carried over from our original actions build. The pyx source file explanation makes sense, I've changed the comment to just say why we use `develop` instead of `install`.